### PR TITLE
ida: add support for COFF and extern functions

### LINKED ISSUE #1193: ida: extractor doesn't recognize calls to extern symbols
When analyzing .lib/COFF files, such as those produced while building a static library, capa doesn't recognize calls to extern symbols. These are calls to routines that will be fixed up by the linker in a subsequent compilation step but aren't available yet. Notably, there is a symbol name associated with the call target. capa should recognize the extern symbol name and emit an API feature for it.

This is probably not super relevant for typical capa workflows, because usually the entire linked executable is not available (we don't even really support analyzing non-PE/elf files). However, for systems that use capa to extract features from functions, this is useful.

I'm not certain of the best way to identify extern symbols in IDA, but one way is to walk the segments and collect those of type `SEG_XTRN`. Any call targets within these sections are extern symbols.

```py
    ... some global setup ...

        extern_ranges = []

        for segstart in idautils.Segments():
            seg = ida_segment.getseg(segstart)

            if not (seg.type == ida_segment.SEG_XTRN):
                continue

            extern_ranges.append(
                (seg.start_ea, seg.end_ea)
            )

    ... extract_insn_api_features ...

        for start, end in extern_ranges:
            if start <= target < end:
                name = ida_name.get_name(target_func.start_ea)
                yield capa.features.insn.API(name), ih.address
```

### PR DESCRIPTION
**Pull request by mike-hunhoff**
close #1193

**note: this impl. emits `format(pe)` for `COFF` files - open to discussion here.**

Screen Shot 2022-12-09 at 12 21 59 PM

### CODE REVIEW

**Review by williballenthin — APPROVED** [review]
thank you!

File: capa/features/extractors/ida/insn.py [hunk 1]
```diff
@@ -70,11 +76,15 @@ def extract_insn_api_features(fh: FunctionHandle, bbh: BBHandle, ih: InsnHandle)
     if not insn.get_canon_mnem() in ("call", "jmp"):
         return
 
-    for api in check_for_api_call(fh.ctx, insn):
-        dll, _, symbol = api.rpartition(".")
-        for name in capa.features.extractors.helpers.generate_symbols(dll, symbol):
+    # check calls to imported functions
+    for api in check_for_api_call(insn, get_imports(fh.ctx)):
```
Comment (mr-tz) [inline capa/features/extractors/ida/insn.py:80]: what's the type here of `api`? worth splitting up or documenting?

File: capa/features/extractors/ida/insn.py [hunk 2]
```diff
@@ -70,11 +76,15 @@ def extract_insn_api_features(fh: FunctionHandle, bbh: BBHandle, ih: InsnHandle)
...
+        for name in capa.features.extractors.helpers.generate_symbols(api[0], api[1]):
             yield API(name), ih.address
 
+    # check calls to extern functions
+    for api in check_for_api_call(insn, get_externs(fh.ctx)):
```
Comment (mr-tz) [inline capa/features/extractors/ida/insn.py:85]: same here (type of api)

**Review by mr-tz — APPROVED** [review]
thanks, one comment, but not a stopper

File: capa/features/extractors/ida/insn.py (hunk 1 above)
Comment (mike-hunhoff) [reply capa/features/extractors/ida/insn.py:80]: I can put a comment here and below to make it more obvious. `check_for_api_call` returns a `tuple`.

**Pull request merged**